### PR TITLE
Integrate AuthSCH login

### LINKED ISSUE #13: Implement AuthSCH authentication.
The gem are needed for this. The first is Devise which can make models authenticatble and provide current_user. The other one in .env whic holds ENV varibles for accessing authSCH.

### PR DESCRIPTION
**Pull request by bbucsy**
- Basically copied from schorpong project
- Integrated AuthSCH with Omniauth gem.
- Added helpers to get if user is logged in, and to get user: example usage can be found in `_nav.html.haml`
- Added user scaffold with basic fields, authorization may need extra fields in the future.
- Added example `.env` file and some description in `Readme.md`
- Designing user views will be another issue.

closes #13

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -62,3 +62,8 @@ gem "haml", "~> 5.1"
 gem 'haml-rails', '~> 2.0', '>= 2.0.1'
 # Use i18n_generators to generate model attribute names automatically
 gem 'i18n_generators', '~> 2.2'
+#use omniauth to integrate AuthSCH
+gem 'omniauth'
+gem 'omniauth-oauth2'
+
+gem 'dotenv-rails', groups: [:development, :test]
```
Comment (SepsiLaszlo) [inline Gemfile:69]: Why do we exclude the production environment?

**Review by SepsiLaszlo — CHANGES_REQUESTED** [review]
Nice work!
It's impressive that how fast you implemented this feature! 🏎️   

I just run throught the PR and found one intresting thing. 
I will look at in in more detail tomorrow.

File: Gemfile (hunk 1 above)
Comment (SepsiLaszlo) [reply Gemfile:69]: Or why do we set development and test?

Comment (bbucsy) [reply Gemfile:69]: Well the github page of the dotenv gem said:
> dotenv was originally created to load configuration variables into ENV in development. There are typically better ways to manage configuration in production environments... 

I put the enviroment variables in the `docker-compose.yml` file, but if we want we can store all aour env variables in a dotenv file and use a `.env.production` file for the production variables.

**Review by SepsiLaszlo — APPROVED** [review]
Good work!

We should create an issue for implementing authorization. 
Some thing we should pay attention to:

- users should only edit their on profile
- uid should not be editable
- users could only edit their own comments

File: Gemfile (hunk 1 above)
Comment (SepsiLaszlo) [reply Gemfile:69]: Good point! 
We should keep it as is.

**Review by triszt4n — APPROVED** [review]
Clean

**Pull request merged**